### PR TITLE
Feature/leaky re lu

### PR DESCRIPTION
**Pull request by sanuj**
#2699
Please review @lisitsyn 
There are more unit-tests to be added. I'll push them later.

### CODE REVIEW

File: src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+void CNeuralLeakyRectifiedLinearLayer::compute_activations(
+		SGVector<float64_t> parameters,
+		CDynamicObjectArray* layers)
+{
+	CNeuralLinearLayer::compute_activations(parameters, layers);
+	
+	int32_t len = m_num_neurons*m_batch_size;
+	for (int32_t i=0; i<len; i++)
+	{
+		m_activations[i] = CMath::max<float64_t>(0.01*m_activations[i], m_activations[i]);
```
Comment (lisitsyn) [inline src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.cpp]: Lets make 0.01 a parameter

**Comment by sanuj** [comment]
@lisitsyn added a parameter `alpha` whose default value is 0.01.

File: src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#define __NEURALLEAKYRECTIFIEDLINEARLAYER_H__
+
+#include <shogun/neuralnets/NeuralRectifiedLinearLayer.h>
+
+namespace shogun
+{
+/** @brief Neural layer with [leaky rectified linear neurons]
+ * (http://en.wikipedia.org/wiki/Rectifier_%28neural_networks%29).
+ * 
+ * Activations are computed according to max(0.01*(W*x+b),W*x+b) where W is the weight
```
Comment (lisitsyn) [inline src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h]: alpha=0.01

File: tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	for (int32_t i=0; i<A_ref.num_rows; i++)
+	{
+		for (int32_t j=0; j<A_ref.num_cols; j++)
+		{
+			A_ref(i,j) = biases[i];
+			
+			for (int32_t k=0; k<x.num_rows; k++)
+				A_ref(i,j) += weights[i+k*A_ref.num_rows]*x(k,j);
+			
+			A_ref(i,j) = CMath::max<float64_t>(0.01*A_ref(i,j), A_ref(i,j));
```
Comment (lisitsyn) [inline tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc]: Alpha

Comment (sanuj) [inline tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc]: @lisitsyn done

**Comment by lisitsyn** [comment]
Ok test failures are not related to these changes.

File: src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	 * @param parameters Vector of size get_num_parameters(), contains the 
+	 * parameters of the layer
+	 * 
+	 * @param layers Array of layers that form the network that this layer is 
+	 * being used with
+	 *
+	 * @param alpha parameter used in calculating max(alpha*x, x)
+	 */
+	virtual void compute_activations(SGVector<float64_t> parameters,
+			CDynamicObjectArray* layers, float64_t alpha=0.01);
```
Comment (lisitsyn) [inline src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h]: Indentation looks wrong here

Comment (lisitsyn) [inline src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h]: Please make alpha a parameter of object (like m_alpha) which defaults to 0.01 but has some getter/setter

File: src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+#include <shogun/neuralnets/NeuralRectifiedLinearLayer.h>
+
+namespace shogun
+{
+/** @brief Neural layer with [leaky rectified linear neurons]
+ * (http://en.wikipedia.org/wiki/Rectifier_%28neural_networks%29).
+ * 
+ * Activations are computed according to max(alpha*(W*x+b),W*x+b) where W is the weight 
+ * matrix, b is the bias vector, x is the input vector, and alpha is a parameter usually
```
Comment (lisitsyn) [inline src/shogun/neuralnets/NeuralLeakyRectifiedLinearLayer.h]: Put the default value of alpha here please ;)

**Comment by lisitsyn** [comment]
Just a few minor issues more and we can merge it

**Comment by sanuj** [comment]
@lisitsyn made the suggested changes ;)

**Comment by lisitsyn** [comment]
Thanks!

**Pull request merged**